packages/amplify-provider-awscloudformation/src/push-resources.ts

```diff
@@ -708,7 +698,16 @@ async function uploadTemplateToS3(
     throw error;
   }
 
-  const templateURL = `https://s3.amazonaws.com/${projectBucket}/amplify-cfn-templates/${category}/${cfnFile}`;
+  return `https://s3.amazonaws.com/${projectBucket}/amplify-cfn-templates/${category}/${cfnFile}`;
+}
+
+function updateCategoryProviderMetaData(
+  context: $TSContext,
+  category: string,
+  resourceName: string,
+  amplifyMeta: $TSMeta,
+  templateURL: string,
+) {
   const providerMetadata = amplifyMeta[category][resourceName].providerMetadata || {};
 
   providerMetadata.s3TemplateURL = templateURL;
```

Comment:
I think `uploadTemplateToS3` will be called for both the auth stack and the new trigger stack which means that the s3TemplateURL will be overwritten by whichever template is uploaded last

Reply:
discussed offline